File: cookbook/python/documents/create_messages.py
import os
from dotenv import load_dotenv
from retab import Retab

# ---------------------------------------------
## Quick example: Parse a document in markdown format using Retab.
# ---------------------------------------------

# Load environment variables
load_dotenv()

retab_api_key = os.getenv("RETAB_API_KEY")

assert retab_api_key, "Missing RETAB_API_KEY"

# Retab Setup
client = Retab(api_key=retab_api_key)
result = client.documents.parse(
    document="../../assets/docs/booking_confirmation.jpg",
    model="gemini-2.5-flash",
    image_resolution_dpi=192,
    browser_canvas="A4",
)

print(result.pages)

File: cookbook/typescript/documents/create_messages.ts
import * as process from 'process';
import { Retab } from '@retab/node';
import { config } from 'dotenv';

async function main() {
  // ---------------------------------------------
  // Quick example: Parse a document in markdown format using Retab.
  // ---------------------------------------------

  // Load environment variables
  config();

  const retabApiKey = process.env.RETAB_API_KEY;

  if (!retabApiKey) {
    throw new Error('Missing RETAB_API_KEY');
  }

  // Retab Setup
  const client = new Retab({ apiKey: retabApiKey });
  // create() takes positional args: (document, model?, tableParsingFormat?, imageResolutionDpi?)
  const result = await client.parses.create(
    '../../../assets/docs/booking_confirmation.jpg',
    'retab-small',
    undefined, // tableParsingFormat
    192, // imageResolutionDpi
  );

  console.log(result.output.pages);


}

main().catch(console.error);
